docs/run-platform-server/configuring.mdx

```diff
@@ -0,0 +1,162 @@
+| `DATABASE_URL`       | postgres://localhost/horizon_pubnet |
+| `INGEST`             | false                               |
+| `TX_SUB_ENABLED`     | false                               |
+
+### Transaction Submission Role Instance
+
+You can run none or multiple instances to serve transaction submission requests. If you run an instance with transaction submission enabled, your deployment is required to perform the ingestion role on the same database referenced here. Horizon transaction submission depends on **live** ingestion to confirm tx submission status.
+
+If ingestion is planned to be done on a separate instance, add `INGEST=false` on this instance, otherwise don't include the parameter, Horizon will default to `INGEST=true` if absent.
+
+If not using the `NETWORK` variable, then must provide the `STELLAR_CORE_URL` variable.
+
+**TODO - Confirm Horizon can use captive core to send tx sub requests and skip stellar-core-url?. **
+
+| environment variable | example                             |
+| -------------------- | ----------------------------------- |
+| `DATABASE_URL`       | postgres://localhost/horizon_pubnet |
+| `NETWORK`            | pubnet                              |
+| `INGEST`             | false                               |
+
+## Notes
+
+- If you have configured your deployment to perform the ingestion role, then it is **strongly** recommended to review [Ingestion](./ingestion.mdx) first and [Filtering](./ingestion-filtering) second and factor that into configuration parameters to achieve best performance related to your application requirements before proceeding further.
+
+- `TX_SUB_ENABLED`, optional, enabled as TRUE by default. Controls whether Horizon will accept HTTP requests to `/tx` API endpoint and forward to network. Refer to [Channel Accounts](../encyclopedia/channel-accounts.mdx) for some recommendations on optional client transaction submission optimizations.
+
+  - When enabled, requires **live** ingestion process to be running on the same database because Horizon depends on new ledgers from network to confirm a transaction submission status, Horizon will report startup error if it detects no **live** ingestion. Requires `NETWORK` or `STELLAR_CORE_URL` be defined for access to network.
+  - When disabled with FALSE, Horizon will return 405 on POSTs to /tx.
+
+- `NETWORK`, optional, can be one of Stellar's public networks, 'pubnet', or 'devnet'. Triggers horizon to automatically set configurations for remaining horizon settings and generate the correct core toml/cfg settings. If you only need Horizon to connect to one of those public Stellar networks, this will take care of all related configuration.
+
+**TODO - create the new `NETWORK` env variable and the functionality it performs to configure all aspects of network connection, including toml, passphrase, archive urls, etc for networks 'pubnet' and 'testnet'. Refer to quickstart for example of same consolidated config parameter for setting network. **
```

Comment:
I'm creating a dev ticket for this task and wanted to clarify a few points
1. If "NETWORK" is not specified, Horizon will still automatically set the values for CAPTIVE_CORE_CONFIG_PATH and CAPTIVE_CORE_STORAGE_PATH.

2. The CAPTIVE_CORE_STORAGE_PATH is not dependent on the network chosen. So the default storage path will remain the same regardless of the network.

Reply:
> If "NETWORK" is not specified, Horizon will still automatically set the values for CAPTIVE_CORE_CONFIG_PATH and CAPTIVE_CORE_STORAGE_PATH.

Do you see that behavior based on `NETWORK-PASSPHRASE`, I found an auto network config feature done in past based on that value, [go#3783](https://github.com/stellar/go/pull/3783), was not aware of this! I think the intent for its usage was more oriented to ease for dev, the design discussion struggled with prod usage, which we should learn from. 

@mollykarcher , I wonder how much this auto-config default behavior of `NETWORK-PASSPHRASE` is being relied upon in the field? It's not mentioned in the -help output, seems kind of quietly in there. Should we focus on shoring up the existing auto-config functionality of `NETWORK-PASSPHRASE` and referring to that usage here in docs or move forward with `NETWORK`? I kinda of prefer the latter as the former seems to conflate the meaning of what appears to be just a value-based setting, but need a check on this.

In either case for auto-config, I don't think Horizon should bother to configure default network or captive core settings if the key network settings are absent, as there is a valid configuration for the API role, where INGEST=false, TX_SUB_ENABLED=false and no network access is required, only the database is used. 

https://github.com/stellar/stellar-docs/pull/162/files/8ba722ce02749aa17849e3bf3a5c1f342a0ebf3a#diff-a2be184a631b87a98857de0dd0abe901235c4639179be7c62719b5ab4e6502ddR51

@urvisavla , this touches on need for new validations on config param settings that we should capture in the ticket also which we propose Horizon to detect/warn about, i.e. if INGEST=true or TX_SUB_ENABLED=true then network config settings must be present, either the key auto-config parameter or the collection of lower level params.